File: 2023/day6/python/p2.py
with open("../input.txt", "r") as f:
    lines = f.read().splitlines()

times = lines[0].split(":")[1].strip().split()
distances = lines[1].split(":")[1].strip().split()

time = int("".join(times))
score = int("".join(distances))
total = 0
print(time)

speed = 1
while speed < time:
    distance = speed * (time - speed)
    if distance > score:
        total += 1

    speed += 1

print(total)

File: 2023/day6/cpp/p2.cpp
#include <iostream>

using namespace std;

int main() {
    unsigned long long time = 40709879;
    unsigned long long score = 215105121471005;
    unsigned int speed = 1;
    int total = 0;
    while (speed < time) {
        unsigned long long distance = speed * (time - speed);
        if (distance > score)
            ++total;
        ++speed;
    }
    cout << total << endl;
}
